mlserver/model.py

```diff
@@ -95,10 +98,20 @@ def decode(
 
         return request_input.data
 
-    def decode_request(self, inference_request: InferenceRequest) -> Any:
-        return decode_inference_request(
-            inference_request, self._settings, self._inputs_index
-        )
+    def decode_request(
+        self,
+        inference_request: InferenceRequest,
+        default_codec: Optional[Union[Type[RequestCodec], RequestCodec]] = None,
+    ) -> Any:
+        decode_inference_request(inference_request, self._settings, self._inputs_index)
+
+        if has_decoded(inference_request):
+            return get_decoded(inference_request)
+
+        if default_codec:
+            return default_codec.decode(inference_request)
+
```

Comment:
why this logic is not more natural to be hidden in `decode_inference_request`?

otherwise could you add a note what are we doing here? as I am a bit confused what `get_decoded` does without having to look at the implementation?

Reply:
I remember there was a reason to have this extra granularity, although I can't say what it was right now. I would leave it as it is for now.